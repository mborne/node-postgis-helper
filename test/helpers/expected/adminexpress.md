# adminexpress

## region

| Name      | Type                          | Required | Description |
| --------- | ----------------------------- | -------- | ----------- |
| id        | `text`                        | N        |             |
| nom_reg   | `text`                        | N        |             |
| nom_reg_m | `text`                        | N        |             |
| insee_reg | `text`                        | N        |             |
| chf_reg   | `text`                        | N        |             |
| geom      | `geometry(MultiPolygon,4326)` | Y        |             |

## departement

| Name      | Type                          | Required | Description |
| --------- | ----------------------------- | -------- | ----------- |
| id        | `text`                        | N        |             |
| nom_dep   | `text`                        | N        |             |
| nom_dep_m | `text`                        | N        |             |
| insee_dep | `text`                        | N        |             |
| insee_reg | `text`                        | N        |             |
| chf_dep   | `text`                        | N        |             |
| geom      | `geometry(MultiPolygon,4326)` | Y        |             |

## commune

| Name       | Type                          | Required | Description |
| ---------- | ----------------------------- | -------- | ----------- |
| id         | `text`                        | N        |             |
| type       | `text`                        | N        |             |
| nom_com    | `text`                        | N        |             |
| nom_com_m  | `text`                        | N        |             |
| insee_com  | `text`                        | N        |             |
| statut     | `text`                        | N        |             |
| population | `numeric(8,0)`                | N        |             |
| insee_arr  | `text`                        | N        |             |
| nom_dep    | `text`                        | N        |             |
| insee_dep  | `text`                        | N        |             |
| nom_reg    | `text`                        | N        |             |
| insee_reg  | `text`                        | N        |             |
| code_epci  | `text`                        | N        |             |
| geom       | `geometry(MultiPolygon,4326)` | Y        |             |
