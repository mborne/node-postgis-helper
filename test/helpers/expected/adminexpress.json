{
  "name": "adminexpress",
  "tables": [
    {
      "name": "region",
      "title": "Région",
      "primaryKey": [
        "id"
      ],
      "foreignKeys": [],
      "columns": [
        {
          "name": "id",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_reg",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_reg_m",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_reg",
          "type": "text",
          "required": false,
          "unique": true,
          "reference": null
        },
        {
          "name": "chf_reg",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "geom",
          "type": "geometry(MultiPolygon,4326)",
          "required": true,
          "unique": false,
          "reference": null
        }
      ]
    },
    {
      "name": "departement",
      "title": "Département",
      "primaryKey": [
        "id"
      ],
      "foreignKeys": [],
      "columns": [
        {
          "name": "id",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_dep",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_dep_m",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_dep",
          "type": "text",
          "required": false,
          "unique": true,
          "reference": null
        },
        {
          "name": "insee_reg",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": {
            "schema": "adminexpress",
            "name": "region",
            "column": "insee_reg"
          }
        },
        {
          "name": "chf_dep",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "geom",
          "type": "geometry(MultiPolygon,4326)",
          "required": true,
          "unique": false,
          "reference": null
        }
      ]
    },
    {
      "name": "commune",
      "title": "Commune",
      "primaryKey": [
        "id"
      ],
      "foreignKeys": [],
      "columns": [
        {
          "name": "id",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "type",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_com",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_com_m",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_com",
          "type": "text",
          "required": false,
          "unique": true,
          "reference": null
        },
        {
          "name": "statut",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "population",
          "type": "numeric(8,0)",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_arr",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "nom_dep",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_dep",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": {
            "schema": "adminexpress",
            "name": "departement",
            "column": "insee_dep"
          }
        },
        {
          "name": "nom_reg",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "insee_reg",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "code_epci",
          "type": "text",
          "required": false,
          "unique": false,
          "reference": null
        },
        {
          "name": "geom",
          "type": "geometry(MultiPolygon,4326)",
          "required": true,
          "unique": false,
          "reference": null
        }
      ]
    }
  ]
}